File: 2501.js
const fs = require("fs");
const [N, K] = fs.readFileSync("/dev/stdin").toString().split(" ").map(Number);
function divisor(N, K) {
  let divisors = [];
  for (let i = 1; i <= N; i++) {
    if (N % i === 0) {
      divisors.push(i);
    }
  }
  if (divisors.length < K) {
    console.log(0);
  } else {
    console.log(divisors[K - 1]);
  }
}

File: 2501.py
# def divisor(N, K):
#   divisors = []
#   i = 1;
#   while i <= N:
#     i = i + 1
#     if N % i == 0:
#       divisors.push(i);
#   if (divisors.length < K):
#     print(0)
#   else:
#     print(divisors[K-1])
N, K = map(int, input().split())
lst = []
for i in range(1, N+1):
  if N % i == 0:
    lst.append(i)

if len(lst) < K :
  print(0)
else :
  print(lst[K-1])